nil.go

```diff
@@ -0,0 +1,64 @@
+package pack
+
+import (
+	"math/rand"
+	"reflect"
+)
+
+// Nil represents a nil value.
+type Nil struct{}
+
+// NewNil wraps an existing raw struct.
+func NewNil(x struct{}) Nil {
+	return Nil(x)
+}
+
+// Type returns nil type.
+func (x Nil) Type() Type {
+	return typeNil{}
+}
+
+// Equal returns true when x is equal to y. Otherwise, it returns false.
+func (x Nil) Equal(y Nil) bool {
+	return x == y
```

Comment:
```suggestion
	return true
```

Reply:
Nil values are always equal to each other, because there is no "room" for variability.